base/threadingconstructs.jl

```diff
@@ -86,15 +85,17 @@ function _threadsfor(iter, lbody, schedule)
             end
         end
         end
-        if ccall(:jl_in_threaded_region, Cint, ()) != 0
+        if $(schedule === :dynamic)
+            threading_run(threadsfor_fun, false)
+        elseif ccall(:jl_in_threaded_region, Cint, ()) != 0
             $(if schedule === :static
               :(error("`@threads :static` cannot be used concurrently or nested"))
               else
               # only use threads when called from outside @threads
-              :(Base.invokelatest(threadsfor_fun, true))
+              :(Base.invokelatest(threadsfor_fun; onethread = true))
```

Comment:
I just realized that we shouldn't use `invokelatest` here. It's not related to this PR so I opened https://github.com/JuliaLang/julia/pull/43957

But you might get merge conflict if #43957 lands first